src/feature/relay/transport_config.c

```diff
@@ -230,6 +230,21 @@ options_validate_server_transport(const or_options_t *old_options,
     char *bindaddr = get_bindaddr_from_transport_listen_line(cl->value, NULL);
     if (!bindaddr)
       REJECT("ServerTransportListenAddr did not parse. See logs for details.");
+
+    tor_addr_t tor_addr;
+    uint16_t tor_port;
+    int rv = tor_addr_port_parse(LOG_WARN, bindaddr, &tor_addr, &tor_port, 0);
+    if (rv < 0) {
+        log_notice(LD_CONFIG, "Your ServerTransportListenAddr failed to parse "
+                              "as a Tor address; check your configuration.");
```

Comment:
This tor_addr_t parsing just accepts IPv4 and IPv6 addresses:
```suggestion
                              "as an IPv4 or IPv6 address; check your configuration.");
```